File: AcademicInfo/src/main/kotlin/domain/OptionalsSelection.kt
package domain

import org.ktorm.entity.Entity
import org.ktorm.schema.Table
import org.ktorm.schema.int

interface OptionalsSelection : Entity<OptionalsSelection> {
    val studentId : Student
    val optionalDisciplineId : OptionalDiscipline
    val priority : Int
}

object OptionalsSelections : Table<OptionalsSelection>("Optionals_Selection") {
    val studentId = int("StudentID").primaryKey().references(Students) { it.studentId }
    val optionalDisciplineId = int("ODisciplineID").primaryKey().references(OptionalDisciplines)
    { it.optionalDisciplineId}
    val priority = int("Priority").bindTo { it.priority }
}

File: academic-info-golang/model/optionals_selection.go
package model

type OptionalsSelection struct {
	StudentId            int `gorm:"primaryKey;column:student_id;type:int"`
	OptionalDisciplineId int `gorm:"primaryKey;column:optional_discipline_id;type:int"`
	Priority             int `gorm:"column:priority;type:int"`
	Student              *Student
	OptionalDiscipline   *OptionalDiscipline
}
